### ISSUE TITLE
Taskbar do not recognize separate chrome instances anymore

### ISSUE THREAD

**Issue by Reddiiiee** [UbuntuBudgie/budgie-desktop-environment#27]
Hello, today i updated budgie but i had several icon on the launcher for google keep, inbox, telegram which are instances of chrome (separate windows of chrome) and the taskbar was working flawlessly: one click open web app, click reduce to icon and so on.

Now, every webapp takes the place of the chrome icon which itself turn into the favicon of the webapp launched. Successive webapps take their own icon.
If chrome is open, then the webapp will start it's own icon, but not the pinned one... so i tried deleting them but now i can't "pin to panel" them anymore. Does this happen only here? 

Steps to replicate: add one or more applications from the chrome store (e.g. inbox) open, chrome app menu, create application shortcut, launch it from the menu or try and pin it.

This is a serious issue for me as i almost use only webapps but Steam, Kodi and Chrome itself. Figured out i ask here 'cause the updates came from budgie-remix ppa but if the issue is from the master budgie repo i can move the thread there as well.
Any reply appreciated... I love budgie's look, but this problem will make my desktop pretty much hard to use.

P.s. sorry for my poor english, i will explain further if i need to.

EDIT forgot to write i'm using budgie remix 16.04 64 bit

**Comment by fossfreedom** [comment]
Please can you add a screenshot of the issue please? 

When you said you updated budgie did you mean that you just installed normal ubuntu updates or did you install new stuff from a ppa - if so what did you update and from where?

**Comment by Reddiiiee** [comment]
Hello, thanks for the fast reply! 
I accepted the update when the system asked me, no ppa added:

```
cat /var/log/apt/history.log

Start-Date: 2016-09-08  15:03:07
Commandline: aptdaemon role='role-commit-packages' sender=':1.82'
Install: python3-notify2:amd64 (0.3-3, automatic), libpyzy-1.0-0v5:amd64 (1.0.1-4.1, automatic), python-xdg:amd64 (0.25-4, automatic), gnome-system-monitor:amd64 (3.18.2-1ubuntu1, automatic), apturl-common:amd64 (0.5.2ubuntu11.1, automatic), libopencc1:amd64 (0.4.3-2build1, automatic), wireless-tools:amd64 (30~pre9-8ubuntu1, automatic), policykit-desktop-privileges:amd64 (0.20, automatic), toshset:amd64 (1.76-4, automatic), apturl:amd64 (0.5.2ubuntu11.1, automatic), nautilus-share:amd64 (0.7.3-2ubuntu1, automatic), gvfs-fuse:amd64 (1.28.2-1ubuntu1~16.04.1, automatic), libxdo3:amd64 (1:3.20150503.1-2, automatic), acpi-support:amd64 (0.142, automatic), ibus-table:amd64 (1.9.1-3ubuntu2, automatic), ibus-pinyin:amd64 (1.5.0-3ubuntu3, automatic), xdotool:amd64 (1:3.20150503.1-2, automatic), thermald:amd64 (1.5-2ubuntu2, automatic), ppa-purge:amd64 (0.2.8+bzr63, automatic), budgie-welcome:amd64 (0.3.9, automatic), libgtkmm-3.0-1v5:amd64 (3.18.0-1, automatic)
Upgrade: libunity-settings-daemon1:amd64 (15.04.1+16.04.20160209-0ubuntu2, 15.04.1+16.04.20160701-0ubuntu1), libnm-glib4:amd64 (1.2.0-0ubuntu0.16.04.2, 1.2.0-0ubuntu0.16.04.3), fuse:amd64 (2.9.4-1ubuntu3, 2.9.4-1ubuntu3.1), budgie-wallpapers:amd64 (16.04.1-1, 16.04.1.1), poppler-utils:amd64 (0.41.0-0ubuntu1, 0.41.0-0ubuntu1.1), budgie-desktop-common:amd64 (0.1-1, 0.2.1-0ubuntu1), libgles2-mesa:amd64 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), libbudgietheme0:amd64 (10.2.5-1ubuntu3, 10.2.6-0ubuntu4), gnome-settings-daemon-schemas:amd64 (3.18.2-0ubuntu3, 3.18.2-0ubuntu3.1), gnome-session:amd64 (3.18.1.2-1ubuntu1.16.04.1, 3.18.1.2-1ubuntu1.16.04.2), language-pack-gnome-en:amd64 (1:16.04+20160415, 1:16.04+20160627), language-pack-gnome-it:amd64 (1:16.04+20160415, 1:16.04+20160627), unity-control-center-faces:amd64 (15.04.0+16.04.20160413-0ubuntu3, 15.04.0+16.04.20160705-0ubuntu1), libdrm-nouveau2:amd64 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), libdrm-nouveau2:i386 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), gnome-session-common:amd64 (3.18.1.2-1ubuntu1.16.04.1, 3.18.1.2-1ubuntu1.16.04.2), gir1.2-gtk-3.0:amd64 (3.18.9-1ubuntu3, 3.18.9-1ubuntu3.1), gir1.2-gmenu-3.0:amd64 (3.13.3-6ubuntu3, 3.13.3-6ubuntu3.1), xserver-common:amd64 (2:1.18.3-1ubuntu2.2, 2:1.18.3-1ubuntu2.3), python3-software-properties:amd64 (0.96.20.2, 0.96.20.4), openssl:amd64 (1.0.2g-1ubuntu4.1, 1.0.2g-1ubuntu4.2), xserver-xorg-core:amd64 (2:1.18.3-1ubuntu2.2, 2:1.18.3-1ubuntu2.3), libgles1-mesa:amd64 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), update-notifier-common:amd64 (3.168, 3.168.1), gnome-sudoku:amd64 (1:3.18.2-1, 1:3.18.4-0ubuntu2), libsystemd0:amd64 (229-4ubuntu6, 229-4ubuntu7), libsystemd0:i386 (229-4ubuntu6, 229-4ubuntu7), libgtk-3-common:amd64 (3.18.9-1ubuntu3, 3.18.9-1ubuntu3.1), grub-common:amd64 (2.02~beta2-36ubuntu3, 2.02~beta2-36ubuntu3.2), libgtk-3-0:amd64 (3.18.9-1ubuntu3, 3.18.9-1ubuntu3.1), budgie-desktop-environment:amd64 (0.3.5-0ubuntu1, 0.3.10), gstreamer1.0-plugins-good:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), libglapi-mesa:amd64 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), libglapi-mesa:i386 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), python2.7-minimal:amd64 (2.7.11-7ubuntu1, 2.7.12-1~16.04), language-selector-common:amd64 (0.165.3, 0.165.4), p11-kit:amd64 (0.23.2-3, 0.23.2-5~ubuntu16.04.1), libnm-gtk-common:amd64 (1.2.0-0ubuntu0.16.04.3, 1.2.0-0ubuntu0.16.04.4), sudo:amd64 (1.8.16-0ubuntu1.1, 1.8.16-0ubuntu1.2), libpython2.7:amd64 (2.7.11-7ubuntu1, 2.7.12-1~16.04), google-chrome-stable:amd64 (51.0.2704.106-1, 53.0.2785.101-1), python2.7:amd64 (2.7.11-7ubuntu1, 2.7.12-1~16.04), console-setup-linux:amd64 (1.108ubuntu15, 1.108ubuntu15.2), libpython3.5:amd64 (3.5.1-10, 3.5.2-2~16.01), python3.5:amd64 (3.5.1-10, 3.5.2-2~16.01), libxatracker2:amd64 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), gnome-settings-daemon:amd64 (3.18.2-0ubuntu3, 3.18.2-0ubuntu3.1), libnm-glib-vpn1:amd64 (1.2.0-0ubuntu0.16.04.2, 1.2.0-0ubuntu0.16.04.3), python3.5-minimal:amd64 (3.5.1-10, 3.5.2-2~16.01), grub2-common:amd64 (2.02~beta2-36ubuntu3, 2.02~beta2-36ubuntu3.2), udev:amd64 (229-4ubuntu6, 229-4ubuntu7), console-setup:amd64 (1.108ubuntu15, 1.108ubuntu15.2), metacity:amd64 (1:3.18.4-0ubuntu0.3, 1:3.18.7-0ubuntu0.1), libegl1-mesa:amd64 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), gstreamer1.0-plugins-base:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), libnm0:amd64 (1.2.0-0ubuntu0.16.04.2, 1.2.0-0ubuntu0.16.04.3), network-manager-gnome:amd64 (1.2.0-0ubuntu0.16.04.3, 1.2.0-0ubuntu0.16.04.4), language-pack-en:amd64 (1:16.04+20160415, 1:16.04+20160627), libklibc:amd64 (2.0.4-8ubuntu1, 2.0.4-8ubuntu1.16.04.1), language-pack-it:amd64 (1:16.04+20160415, 1:16.04+20160627), upower:amd64 (0.99.4-2ubuntu0.2, 0.99.4-2ubuntu0.3), isc-dhcp-common:amd64 (4.3.3-5ubuntu12, 4.3.3-5ubuntu12.1), network-manager:amd64 (1.2.0-0ubuntu0.16.04.2, 1.2.0-0ubuntu0.16.04.3), software-properties-gtk:amd64 (0.96.20.2, 0.96.20.4), dkms:amd64 (2.2.0.3-2ubuntu11, 2.2.0.3-2ubuntu11.1), gir1.2-budgie-desktop-1.0:amd64 (10.2.5-1ubuntu3, 10.2.6-0ubuntu4), grub-pc:amd64 (2.02~beta2-36ubuntu3, 2.02~beta2-36ubuntu3.2), p11-kit-modules:amd64 (0.23.2-3, 0.23.2-5~ubuntu16.04.1), p11-kit-modules:i386 (0.23.2-3, 0.23.2-5~ubuntu16.04.1), libgstreamer-plugins-good1.0-0:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), libudev1:amd64 (229-4ubuntu6, 229-4ubuntu7), libudev1:i386 (229-4ubuntu6, 229-4ubuntu7), libapparmor1:amd64 (2.10.95-0ubuntu2, 2.10.95-0ubuntu2.2), unity-control-center:amd64 (15.04.0+16.04.20160413-0ubuntu3, 15.04.0+16.04.20160705-0ubuntu1), gtk2-engines-murrine:amd64 (0.98.2-0ubuntu2, 0.98.2-0ubuntu2.1), gstreamer1.0-pulseaudio:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), libgbm1:amd64 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), libllvm3.8:amd64 (1:3.8-2ubuntu3, 1:3.8-2ubuntu4), libllvm3.8:i386 (1:3.8-2ubuntu3, 1:3.8-2ubuntu4), libnm-util2:amd64 (1.2.0-0ubuntu0.16.04.2, 1.2.0-0ubuntu0.16.04.3), budgie-wallpapers-xenial:amd64 (16.04.1-1, 16.04.1.1), grub-pc-bin:amd64 (2.02~beta2-36ubuntu3, 2.02~beta2-36ubuntu3.2), libdrm-amdgpu1:amd64 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), libdrm-amdgpu1:i386 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), cups-filters:amd64 (1.8.3-2ubuntu3, 1.8.3-2ubuntu3.1), libbudgie-plugin0:amd64 (10.2.5-1ubuntu3, 10.2.6-0ubuntu4), libapparmor-perl:amd64 (2.10.95-0ubuntu2, 2.10.95-0ubuntu2.2), libgtk-3-bin:amd64 (3.18.9-1ubuntu3, 3.18.9-1ubuntu3.1), python3-distupgrade:amd64 (1:16.04.14, 1:16.04.16), budgie-remix-lightdm-theme:amd64 (0.4.2, 0.4.5), libraven0:amd64 (10.2.5-1ubuntu3, 10.2.6-0ubuntu4), ubuntu-release-upgrader-core:amd64 (1:16.04.14, 1:16.04.16), metacity-common:amd64 (1:3.18.4-0ubuntu0.3, 1:3.18.7-0ubuntu0.1), libfuse2:amd64 (2.9.4-1ubuntu3, 2.9.4-1ubuntu3.1), libcupsfilters1:amd64 (1.8.3-2ubuntu3, 1.8.3-2ubuntu3.1), language-pack-gnome-en-base:amd64 (1:16.04+20160415, 1:16.04+20160627), systemd-sysv:amd64 (229-4ubuntu6, 229-4ubuntu7), xserver-xorg-video-intel:amd64 (2:2.99.917+git20160325-1ubuntu1, 2:2.99.917+git20160325-1ubuntu1.1), libwayland-egl1-mesa:amd64 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), libmetacity-private3a:amd64 (1:3.18.4-0ubuntu0.3, 1:3.18.7-0ubuntu0.1), libpam-systemd:amd64 (229-4ubuntu6, 229-4ubuntu7), libgstreamer-plugins-base1.0-0:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), libgail-3-0:amd64 (3.18.9-1ubuntu3, 3.18.9-1ubuntu3.1), libdrm2:amd64 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), libdrm2:i386 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), libnma0:amd64 (1.2.0-0ubuntu0.16.04.3, 1.2.0-0ubuntu0.16.04.4), gstreamer1.0-x:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), systemd:amd64 (229-4ubuntu6, 229-4ubuntu7), libnma-common:amd64 (1.2.0-0ubuntu0.16.04.3, 1.2.0-0ubuntu0.16.04.4), language-pack-gnome-it-base:amd64 (1:16.04+20160415, 1:16.04+20160627), libfontembed1:amd64 (1.8.3-2ubuntu3, 1.8.3-2ubuntu3.1), gir1.2-gst-plugins-base-1.0:amd64 (1.8.1-1ubuntu0.1, 1.8.2-1ubuntu0.1), libgl1-mesa-dri:amd64 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), libgl1-mesa-dri:i386 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), libupower-glib3:amd64 (0.99.4-2ubuntu0.2, 0.99.4-2ubuntu0.3), libfwupd1:amd64 (0.7.0-0ubuntu4.2, 0.7.0-0ubuntu4.3), libgnome-menu-3-0:amd64 (3.13.3-6ubuntu3, 3.13.3-6ubuntu3.1), libosmesa6:amd64 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), libosmesa6:i386 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), keyboard-configuration:amd64 (1.108ubuntu15, 1.108ubuntu15.2), gnome-session-bin:amd64 (3.18.1.2-1ubuntu1.16.04.1, 3.18.1.2-1ubuntu1.16.04.2), libnm-gtk0:amd64 (1.2.0-0ubuntu0.16.04.3, 1.2.0-0ubuntu0.16.04.4), language-pack-en-base:amd64 (1:16.04+20160415, 1:16.04+20160627), cups-filters-core-drivers:amd64 (1.8.3-2ubuntu3, 1.8.3-2ubuntu3.1), libgl1-mesa-glx:amd64 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), libgl1-mesa-glx:i386 (11.2.0-1ubuntu2.1, 11.2.0-1ubuntu2.2), ubuntu-release-upgrader-gtk:amd64 (1:16.04.14, 1:16.04.16), libunity-control-center1:amd64 (15.04.0+16.04.20160413-0ubuntu3, 15.04.0+16.04.20160705-0ubuntu1), libpython2.7-minimal:amd64 (2.7.11-7ubuntu1, 2.7.12-1~16.04), libdrm-intel1:amd64 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), libdrm-intel1:i386 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), unity-settings-daemon:amd64 (15.04.1+16.04.20160209-0ubuntu2, 15.04.1+16.04.20160701-0ubuntu1), apparmor:amd64 (2.10.95-0ubuntu2, 2.10.95-0ubuntu2.2), budgie-core:amd64 (10.2.5-1ubuntu3, 10.2.6-0ubuntu4), libp11-kit0:amd64 (0.23.2-3, 0.23.2-5~ubuntu16.04.1), libp11-kit0:i386 (0.23.2-3, 0.23.2-5~ubuntu16.04.1), gir1.2-gstreamer-1.0:amd64 (1.8.1-1~ubuntu1, 1.8.2-1~ubuntu1), libdrm-radeon1:amd64 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), libdrm-radeon1:i386 (2.4.67-1ubuntu0.16.04.1, 2.4.67-1ubuntu0.16.04.2), language-pack-it-base:amd64 (1:16.04+20160415, 1:16.04+20160627), mesa-vdpau-drivers:amd64 (11.2.0-1ubuntu2, 11.2.0-1ubuntu2.2), libpython3.5-stdlib:amd64 (3.5.1-10, 3.5.2-2~16.01), linux-firmware:amd64 (1.157.2, 1.157.3), libpython2.7-stdlib:amd64 (2.7.11-7ubuntu1, 2.7.12-1~16.04), libpoppler-glib8:amd64 (0.41.0-0ubuntu1, 0.41.0-0ubuntu1.1), libpython3.5-minimal:amd64 (3.5.1-10, 3.5.2-2~16.01), libpoppler58:amd64 (0.41.0-0ubuntu1, 0.41.0-0ubuntu1.1), isc-dhcp-client:amd64 (4.3.3-5ubuntu12, 4.3.3-5ubuntu12.1), klibc-utils:amd64 (2.0.4-8ubuntu1, 2.0.4-8ubuntu1.16.04.1), arc-firefox-theme:amd64 (47.20160607-1ubuntu1, 48.20160802-0ubuntu1), update-notifier:amd64 (3.168, 3.168.1), libgstreamer1.0-0:amd64 (1.8.1-1~ubuntu1, 1.8.2-1~ubuntu1), ubuntu-mono:amd64 (14.04+16.04.20160621-0ubuntu1, 14.04+16.04.20160804-0ubuntu1), budgie-desktop:amd64 (10.2.5-1ubuntu3, 10.2.6-0ubuntu4), lightdm:amd64 (1.18.1-0ubuntu1, 1.18.2-0ubuntu2), less:amd64 (481-2.1, 481-2.1ubuntu0.1), liblightdm-gobject-1-0:amd64 (1.18.1-0ubuntu1, 1.18.2-0ubuntu2), cups-browsed:amd64 (1.8.3-2ubuntu3, 1.8.3-2ubuntu3.1), base-files:amd64 (9.4ubuntu4.1, 9.4ubuntu4.2), libpanel-applet0:amd64 (1:3.18.2-1ubuntu1, 1:3.18.3-0ubuntu0.1), gnome-menus:amd64 (3.13.3-6ubuntu3, 3.13.3-6ubuntu3.1), software-properties-common:amd64 (0.96.20.2, 0.96.20.4)
End-Date: 2016-09-08  15:05:33
```

Here are the screenshots:
- Before today update the taskbar looks ok, Chrome has it's own icon and webapps too are pinned. Regardless which i have opened first, every webapp (or Chrome windows) has it's own working button:
  ![before](https://cloud.githubusercontent.com/assets/12221659/18367318/e52e6432-761a-11e6-9012-b50dc21e54eb.png)
- After today update, if i launch any of the webapp (you can get the same from the terminal by running
  `google-chrome --app=URL` 
  and Google start a windows with the url given, with no toolbar) then the webapp favicon takes the place of the Chrome launcher itself:
  ![after_substitution](https://cloud.githubusercontent.com/assets/12221659/18367368/10d9e07a-761b-11e6-8599-ebb9816c5ddd.png)

You can see that the first icon is Chrome here and on the previous screenshot it is replaced by Inbox: 
![after_only_chrome](https://cloud.githubusercontent.com/assets/12221659/18367472/97664778-761b-11e6-8d04-02ac81e9ac6c.png)
- However, if i first launch Chrome then the webapp, it looks ok, it has it's own favicon as a button and i can click them to minimize it, but i can no longer pin to the panel:
  ![after_looks_ok](https://cloud.githubusercontent.com/assets/12221659/18367492/b66b240e-761b-11e6-80dc-9ecd10819b78.png)

So basically, my solution would be: Open chrome even if i don't need it to browse, leaving it open, then use Budgie menu to open the web app i need... pretty uncomfortable :(

Maybe the webapp .desktop files needs some manual tweak? It wasn't needed before, i just right click on Chrome app page "create a shortcut to application menu" and voilà. I would be really glad to have that beahivour back! Thanks again for any suggestion!

**Comment by fossfreedom** [comment]
hmm - if you dont mind try and reset your panel configuration.  Launch budgie-welcome and use the "default settings" button and then click "reset budgie panel".

**Comment by fossfreedom** [comment]
as an aside - I know there was a chrome update in the last day or so - maybe chrome itself is the issue here.

**Comment by Reddiiiee** [comment]
Tried resetting, still the same.
I was thinking the same, and in the log there's a Chrome update in fact! I'm not sure how they messed up at this point... Should i move and ask Chrome dev team or maybe Budgie devs?

Thanks for your help

**Comment by fossfreedom** [comment]
hmm - not sure.  Launch "plank" - and copy the desktop shortcut into the plank launcher.  Does this work correctly?

**Comment by Reddiiiee** [comment]
Yes, first thing tried. Apparently it worked with chrome and web apps

**Comment by fossfreedom** [comment]
K ... Since web apps work correctly via plank then this looks very much like an upstream issue via the solus budgie desktop github repo.

**Comment by Reddiiiee** [comment]
I will try there tomorrow asap, thanks for your efforts!

**Comment by Reddiiiee** [comment]
Solved: https://github.com/solus-project/budgie-desktop/issues/566

When we could expect the update on Budgie Remix? Is this github already updated with the fix?

**Comment by fossfreedom** [comment]
@Redneckita v10.2.7 of budgie-desktop is being prepared by upstream.  Given that is it being made available soon, we'll test with that and then push this out as a release for both 16.04 and 16.10 users.

**Comment by Reddiiiee** [comment]
Awesome, thanks!

**Comment by fossfreedom** [comment]
https://plus.google.com/106459857019142382817/posts/2zmG9A5Nws4

**Comment by Reddiiiee** [comment]
Yay! Thanks!
